src/Compilers/Core/Portable/Operations/ControlFlowGraphBuilder.InterpolatedStringContext.cs

```diff
@@ -0,0 +1,108 @@
+            private readonly IOperation? _singleAllowedCreation;
+            private readonly byte _flags;
+            private readonly int _outPlaceholder;
+            private readonly int _handlerPlaceholder;
+
+            public readonly int StartingStackDepth;
+
+            public InterpolatedStringHandlerContext(ImmutableArray<IArgumentOperation> allowedCreations, int startingStackDepth, bool hasReceiver)
+            {
+                _allowedCreations = allowedCreations;
+                _singleAllowedCreation = null;
+                StartingStackDepth = startingStackDepth;
+                _outPlaceholder = -1;
+                _handlerPlaceholder = -1;
+                _flags = (byte)((hasReceiver ? HasReceiverFlag : 0) | IsNotDefaultFlag);
+            }
+
+            public InterpolatedStringHandlerContext(IOperation allowedCreation, int outPlaceholder, int handlerPlaceholder)
+            {
+                _allowedCreations = default;
+                _singleAllowedCreation = allowedCreation;
+                StartingStackDepth = -1;
+                _outPlaceholder = outPlaceholder;
+                _flags = IsNotDefaultFlag;
+                _handlerPlaceholder = handlerPlaceholder;
+            }
+
+            public bool IsDefault => (_flags & IsNotDefaultFlag) == 0;
+            public bool HasReceiver => (_flags & HasReceiverFlag) == HasReceiverFlag;
+
+            public int OutPlaceholder
+            {
+                get => _outPlaceholder;
+                init
+                {
+                    Debug.Assert(!IsDefault);
+                    _outPlaceholder = value;
+                }
+            }
+
+            public int HandlerPlaceholder
+            {
+                get => _handlerPlaceholder;
+                init
+                {
+                    Debug.Assert(!IsDefault);
+                    _handlerPlaceholder = value;
+                }
+            }
+
+            public IOperation SingleAllowedCreation
+            {
+                init
+                {
+                    Debug.Assert(!IsDefault);
+                    _allowedCreations = default;
+                    _singleAllowedCreation = value;
+                }
+            }
+
+            public bool IsCreationAllowed(IInterpolatedStringHandlerCreationOperation operation)
+            {
+                if (IsDefault)
+                {
+                    return false;
+                }
+
+                Debug.Assert((_singleAllowedCreation == null) ^ _allowedCreations.IsDefault);
+                return _singleAllowedCreation != null
+                            ? _singleAllowedCreation == operation
+                            : _allowedCreations.Any(static (argument, operation) => argument.Value == operation, operation);
+            }
+        }
+
+        private static void AssertContainingContextIsForThisCreation(IOperation placeholderOperation, InterpolatedStringHandlerContext context)
```

Comment:
>InterpolatedStringHandlerContext context [](http://example.com/codeflow?start=102&length=40)

It looks like we allways pass ```_currentInterpolatedStringHandlerContext``` for this parameter. What is the benefit in having the parameter vs. making this an instance method.